tests/test-rest-nav-menu-items-controller.php

```diff
@@ -23,85 +45,760 @@ public static function wpSetUpBeforeClass( $factory ) {
+					$link['attributes'] = $attributes;
+				}
+			}
+
+			$this->check_menu_item_data( $menu_item, $data, $context, $links );
+		}
+	}
+
+	/**
+	 * @param $post
+	 * @param $data
+	 * @param $context
+	 * @param $links
+	 */
+	protected function check_menu_item_data( $post, $data, $context, $links ) {
+		$post_type_obj = get_post_type_object( self::POST_TYPE );
+
+		// Standard fields
+		$this->assertEquals( $post->ID, $data['id'] );
+		$this->assertEquals( wpautop( $post->post_content ), $data['description'] );
+
+		// Check filtered values.
+		if ( post_type_supports( self::POST_TYPE, 'title' ) ) {
+			add_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
+			$this->assertEquals( $post->title, $data['title']['rendered'] );
+			remove_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
+			if ( 'edit' === $context ) {
+				$this->assertEquals( $post->post_title, $data['title']['raw'] );
+			} else {
+				$this->assertFalse( isset( $data['title']['raw'] ) );
+			}
+		} else {
+			$this->assertFalse( isset( $data['title'] ) );
+		}
+
+		// post_parent
+
+		$this->assertArrayHasKey( 'parent', $data );
+		if ( $post->post_parent ) {
+			if ( is_int( $data['parent'] ) ) {
+				$this->assertEquals( $post->post_parent, $data['parent'] );
+			} else {
+				$this->assertEquals( $post->post_parent, $data['parent']['id'] );
+				$menu_item = wp_setup_nav_menu_item( get_post( $data['parent']['id'] ) );
+				$this->check_get_menu_item_response( $data['parent'], $menu_item, 'view-parent' );
+			}
+		} else {
+			$this->assertEmpty( $data['parent'] );
+		}
+
+		// page attributes
+		$this->assertEquals( $post->menu_order, $data['menu_order'] );
+
+		$taxonomies = wp_list_filter( get_object_taxonomies( self::POST_TYPE, 'objects' ), array( 'show_in_rest' => true ) );
+		foreach ( $taxonomies as $taxonomy ) {
+			$this->assertTrue( isset( $data[ $taxonomy->rest_base ] ) );
+			$terms = wp_get_object_terms( $post->ID, $taxonomy->name, array( 'fields' => 'ids' ) );
+			sort( $terms );
+			sort( $data[ $taxonomy->rest_base ] );
+			$this->assertEquals( $terms, $data[ $taxonomy->rest_base ] );
+		}
+
+		// test links
+		if ( $links ) {
+			$links = test_rest_expand_compact_links( $links );
+			$this->assertEquals( $links['self'][0]['href'], rest_url( 'wp/v2/' . $post_type_obj->rest_base . '/' . $data['id'] ) );
+			$this->assertEquals( $links['collection'][0]['href'], rest_url( 'wp/v2/' . $post_type_obj->rest_base ) );
+			$this->assertEquals( $links['about'][0]['href'], rest_url( 'wp/v2/types/' . self::POST_TYPE ) );
+
+			$num = 0;
+			foreach ( $taxonomies as $key => $taxonomy ) {
+				$this->assertEquals( $taxonomy->name, $links['https://api.w.org/term'][ $num ]['attributes']['taxonomy'] );
+				$this->assertEquals( add_query_arg( 'post', $data['id'], rest_url( 'wp/v2/' . $taxonomy->rest_base ) ), $links['https://api.w.org/term'][ $num ]['href'] );
+				$num ++;
+			}
+		}
+	}
+
+	/**
+	 * @param $response
```

Comment:
Type docs here as well would be nice.

Reply:
Fixed in https://github.com/WP-API/menus-endpoints/pull/41/commits/565a06fff290cddae77b2167a2d05b457e85e1d6